core/net.go

```diff
@@ -75,6 +81,27 @@ func (n *OpenBazaarNode) SendOfflineMessage(p peer.ID, k *libp2p.PubKey, m *pb.M
 	if err != nil {
 		return err
 	}
+
+	// Post provider to gateway if we have one set in the config
+	if len(n.CrosspostGateways) > 0 {
+		pmes := dhtpb.NewMessage(dhtpb.Message_ADD_PROVIDER, pointer.Cid.KeyString(), 0)
+		pmes.ProviderPeers = dhtpb.RawPeerInfosToPBPeers([]ps.PeerInfo{pointer.Value})
+		ser, err := proto.Marshal(pmes)
+		if err != nil {
+			for _, g := range n.CrosspostGateways {
+				go func() {
+					dial := gonet.Dial
+					if n.TorDialer != nil {
+						dial = n.TorDialer.Dial
+					}
+					tbTransport := &http.Transport{Dial: dial}
+					client := &http.Client{Transport: tbTransport, Timeout: time.Minute}
+					client.Post(g.String()+"ipfs/providers", "application/x-www-form-urlencoded", bytes.NewReader(ser))
+				}()
```

Comment:
because the goroutine is async the value of `g` likely won't be what you expect. All the schedule goroutines could end up using the same value for `g` (the last one).

Make the func take g as a param so it's copied to its stack.